lib/active_merchant/billing/gateways/commerce_hub.rb

```diff
@@ -291,7 +291,9 @@ def headers(request, options)
           'Auth-Token-Type' => 'HMAC',
           'Content-Type' => 'application/json',
           'Accept' => 'application/json',
-          'Authorization' => signature
+          'Authorization' => signature,
+          'x-originator' => 'CommerceHub-Partners-Spreedly',
+          'user-agent' => 'CommerceHub-Partners-Spreedly-V1.00'
```

Comment:
⚠️ Warning ⚠️ 

There is an issue with this approach => See, Spreedly is not the only user of this gem so any other protect/library that depends on this adapter is going to send traffic is if it was Spreedly:

Alternative approach:

1) just add and extra option that will have a hash with the custom headers, something like:

```ruby
options.fetch(:custom_headers, {}).merge({
  'Client-Request-Id' => client_request_id,
  'Api-Key' => @options[:api_key],
  'Timestamp' => time,
  'Accept-Language' => 'application/json',
  'Auth-Token-Type' => 'HMAC',
  'Content-Type' => 'application/json',
  'Accept' => 'application/json',
  'Authorization' => signature
})
```

2) in core you only need to change the CommerceHubGateway#auth_purchase_params to include those headers in the list of params to AM.